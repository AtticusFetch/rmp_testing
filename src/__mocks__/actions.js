export const fetchSome = jest.fn().mockReturnValue({
  then: cb => {
    cb('some mocked data');
    return {
      catch: () => {},
    };
  },
});