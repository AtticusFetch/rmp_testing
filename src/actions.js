export const fetchSome = arg => new Promise((resolve, reject) => {
  setTimeout(() => resolve('some data'), 1000);
});