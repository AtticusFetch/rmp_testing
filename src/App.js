import React, { Component } from 'react';

import SomeComp from './Component';
import { fetchSome } from './actions';

import './App.css';

class App extends Component {
  state = {
    timeoutDone: false,
    data: null,
    isFetched: false,
    error: null,
  };

  componentDidMount() {
    this._timeOutMethod();
  }

  componentWillUpdate(nextProps, nextState) {
    console.group('componentWillUpdate');
    console.log('old state', this.state);
    console.log('new state', nextState);
    console.groupEnd();
  }

  _asyncMethod = () => {
    fetchSome(this.state.data)
      .then(fetchedData => this.setState({ data: fetchedData, isFetched: true }))
      .catch(error => this.setState({ error }));
  };

  _setData = event => this.setState({ data: event.target.value });

  _timeOutMethod = () => {
    setTimeout(() => this.setState({ timeoutDone: true }), 2000);
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <br />
        <button onClick={this._asyncMethod}>Fetch Data</button>
        <input onChange={this._setData} />
        <SomeComp value="somp_value" />
        <div>
          <br />
          <span id="data">Data: </span>{this.state.data && <span>{this.state.data}</span>}
        </div>
        <br />
        <div>
          <span id="isfetched">isFetched: {this.state.isFetched.toString()}</span>
        </div>
        <br />
        <div>
          <span id="timeout">timeoutDone: {this.state.timeoutDone.toString()}</span>
        </div>
      </div>
    );
  }
}

export default App;
