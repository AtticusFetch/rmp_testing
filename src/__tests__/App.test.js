import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { fetchSome } from './../actions';

import App from './../App';

jest.useFakeTimers();

jest.mock('./../actions');
jest.mock('./../Component', () => 'Component');

let props = {};

describe('App', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<App />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should change data', () => {
    const eventMock = {
      target: {
        value: 'test_value'
      }
    };
    const component = shallow(<App />);

    component.find('input').simulate('change', eventMock);

    const textValue = component
      .find('#data + span')
      .text();

      expect(textValue).toEqual(eventMock.target.value);
  });

  it('should set timeoutDone', () => {
    const component = shallow(<App />);

    jest.runAllTimers();
    component.update();

    expect(component.find('#timeout').text()).toMatch(/true/gi);
  });

  it('fetch data', () => {
    const component = shallow(<App {...props} />);

    component.find('button').simulate('click');

    const textValue = component
    .find('#data + span')
    .text();

    expect(textValue).toEqual('some mocked data');
  });
});
