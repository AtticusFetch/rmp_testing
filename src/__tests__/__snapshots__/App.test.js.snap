// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <br />
  <button
    onClick={[Function]}
  >
    Fetch Data
  </button>
  <input
    onChange={[Function]}
  />
  <Component
    value="some_value"
  />
  <div>
    <br />
    <span
      id="data"
    >
      Data: 
    </span>
  </div>
  <br />
  <div>
    <span
      id="isfetched"
    >
      isFetched: 
      false
    </span>
  </div>
  <br />
  <div>
    <span
      id="timeout"
    >
      timeoutDone: 
      false
    </span>
  </div>
</div>
`;
